### ISSUE TITLE
The IOPS value calculated by Iotune is relatively low compared to the NVMe disk used.

### ISSUE THREAD

**Issue by libo-sober** [scylladb/scylladb#14480]
*Installation details*
Scylla version (or git commit hash): 5.1.10
Cluster size: 18 nodes (6 nodes per az) 
OS (RHEL/CentOS/Ubuntu/AWS AMI): centos

*Hardware details (for performance issues)*          Delete if unneeded
Platform (physical/VM/cloud instance type/docker): docker
Hardware: 64cpu / 400G memory
Disks: (SSD/HDD, count): nvme 

When deploying the scylla cluster, `seastar iotune` calculates the value in `io_properties.yaml`. During the pressure test, we found that the latency was high, but at this time, the disk was not fully utilized. The reason is that the IO queue length is set too small, causing the request to block.

The hardware resources we use are better than `aws i3.xlarge`, but the values in `io_properties.yaml` are significantly lower.
![aws i3.xlarge](https://github.com/scylladb/scylladb/assets/70140794/63e3d80f-7c59-429b-acf1-241af2380355)
![calculated iops](https://github.com/scylladb/scylladb/assets/70140794/57208e7a-8928-48d3-b2df-ef3139ce3049)

So what is the reason for the low value calculated `io_properties.yaml`?

We are considering whether it is related to the file system and block size. We see that Scylla expects smaller block size to allow smaller commitlog writes without stalling.
![block size](https://github.com/scylladb/scylladb/assets/70140794/e2825e68-aecc-4929-9c1c-179fefcaa2ad)

However, we compared the block sizes of 1024 and 4096, and found that it did not have a significant impact.
![bsize=1024](https://github.com/scylladb/scylladb/assets/70140794/71fabe66-209f-423a-87d8-2e48c61acef8)
![bsize=4096](https://github.com/scylladb/scylladb/assets/70140794/3d194966-2bd4-402c-adb9-7191b9250d85)

How can we solve this problem? Thanks.

**Comment by mykaul** [comment]
1. See https://github.com/scylladb/seastar/issues/1698 (duplicate?)
2. Scylla does not aim to saturate the disk's bandwidth, as it is optimized to latency.

**Comment by mykaul** [comment]
@libo-sober - what IO queue was not set properly? (I'm also wondering about the huge difference in sequential read performance...

**Comment by libo-sober** [comment]
> 1. See [iotune: Random IO buffer size is not always correct seastar#1698](https://github.com/scylladb/seastar/issues/1698) (duplicate?)
> 2. Scylla does not aim to saturate the disk's bandwidth, as it is optimized to latency.

Sorry, I think I may not have described this issue clearly. There are two main issues here:
1. The value of `io_properties.yaml` calculated through `seastar iotune` is much lower than the IOPS of the actual hard drive.
2. After manually increasing the value of `io_properties.yaml`, I will get a higher IOPS in the test. And it can maintain the same latency.
Where should I make beneficial changes in the face of this situation? Thanks.
In addition, the configuration of disk block size is just my guess (it seems unrelated to this issue). Maybe I'm heading in the wrong direction.

**Comment by xemul** [comment]
@libo-sober , you're correct in that iotune miscalculated write iops and as @mykaul points, we acknowledge there's a problem and there's a issue (scylladb/seastar#1698). However, Scylla exploits disk read-IOPS and write-bandwidth (and, probably, read-bandwidth up to some extent, but I never saw nodes with saturated read bandwidth). Miscalculated write-IOPS should have low effect on Scylla's IO.

The fix should come to seastar in the context of the aforementioned issue

**Comment by libo-sober** [comment]
> Miscalculated write-IOPS should have low effect on Scylla's IO.

We use [go-ycsb](https://github.com/pingcap/go-ycsb) for iops testing. Under the same conditions, first use the `io_properties` value calculated by `iotune ` to test, and then double the value of `io_properties`. The results are as follows:

<html>
<body>


  |   | ops | Avg(ms)
-- | -- | -- | --
iotune | write | 2699.7 | 5.278
  | read | 2699.0 | 5.275
double | write | 3048.5 | 4.650
  | read | 3184.1 | 4.445


</body>
</html>

We can see that the value of ops has increased by approximately 500. May I ask if this meets expectations? Thanks.
We are using a very small SSD disk. If the disk performance is stronger, will there be a greater improvement?

**Comment by mykaul** [comment]
> > Miscalculated write-IOPS should have low effect on Scylla's IO.
> 
> We use [go-ycsb](https://github.com/pingcap/go-ycsb) for iops testing. Under the same conditions, first use the `io_properties` value calculated by `iotune ` to test, and then double the value of `io_properties`. The results are as follows:
> 
>  	 	ops	Avg(ms)
> iotune	write	2699.7	5.278
>  	read	2699.0	5.275
> double	write	3048.5	4.650
>  	read	3184.1	4.445
> We can see that the value of ops has increased by approximately 500. May I ask if this meets expectations? Thanks. We are using a very small SSD disk. If the disk performance is stronger, will there be a greater improvement?

Keep in mind that we do not want to stress the disk to its max bandwidth (interesting  to see avg latency going down in your numbers!) , as it'll impact latency. What are your P99 latency numbers?

**Comment by libo-sober** [comment]
I'm very sorry, we didn't record this value at the time. This may be a special phenomenon (we haven't measured it many times).
There is a question: If we increase this parameter to bring more bandwidth to the disk, what impact will it have on scylladb?
